onnx/external_data_helper.py

```diff
@@ -134,17 +141,25 @@ def convert_model_to_external_data(
                     "location must be a relative path that is relative to the model path."
                 )
             file_name = location
+        current_offset = 0
         for tensor in tensors:
+            tensor_size = len(tensor.raw_data)
             if (
                 tensor.HasField("raw_data")
-                and sys.getsizeof(tensor.raw_data) >= size_threshold
+                and tensor_size >= size_threshold
             ):
-                set_external_data(tensor, file_name)
+                if align_offset and tensor_size > align_threshold:
+                    # Align to the larger of the page size or the allocation granularity
+                    alignment_factor = max(4096, allocation_granularity)  # Assuming 4k page size
+                    current_offset = (current_offset + alignment_factor - 1) // alignment_factor * alignment_factor # Align to the next page or alloc granularity boundary
```

Comment:
Possible to extract this logic into a private function and create a test for it?

Reply:
Agree we should have a test. However, I am not sure I have the bandwidth. Is this something that someone from ONNX could sponsor / contribute code to @justinchuby. yourself or someone else?

Note that @pranavsharma suggested other changes. Such as passing in the OS "Windows, UNIX, MacOsX" as string parameter and determining page size and VirtualAlloc memory allocation granularity based on the OS string

Reply:
Ideally to me the model file should be os agnostic. Picking the least common multiple in this case as default makes sense to me